packages/@okta/vuepress-site/docs/guides/authenticators-google-authenticator/main/java/enrollmentintegrationsteps.md

```diff
@@ -0,0 +1,99 @@
+// TODO: add device context
+
+// set the user's credentials
+AuthenticationOptions authenticationOptions = new AuthenticationOptions(username, password);
+
+// start authentication flow
+AuthenticationResponse authenticationResponse = idxAuthenticationWrapper.authenticate(authenticationOptions, proceedContext);
+```
+
+### 3: Handle the response from the sign-in flow
+
+The `AuthenticationResponse` contains the current state of the authentication process, it could be a `SUCCESS` or may indicate more information is needed to complete the flow, for example, require an additional factor.
+
+Query the `authenticationStatus` property of the `AuthenticationResponse` object returned by `authenticate` to discover the current status of the authentication process.
+
+```java
+// Update the context each time a request is made to Okta
+// make sure it is persisted for web applications
+proceedContext = authenticationResponse.getProceedContext();
+
+// get the authenticationStatus
+AuthenticationStatus authenticationStatus = authenticationResponse.getAuthenticationStatus();
+
+switch (authenticationStatus) {
+    case SUCCESS:
+        … your code …
+    case PASSWORD_EXPIRED:
+        … your code …
+}
+```
+
+### 4: Select a possible authenticator factor
+
+If you configured your Okta org correctly, you need to respond to two specific authenticator statuses to handle this scenario in addition to `SUCCESS` and `PASSWORD_EXPIRED`:
+
+* `AWAITING_PROFILE_ENROLLMENT` that is covered in this section
+* `AWAITING_AUTHENTICATOR_ENROLLMENT_SELECTION` that is covered in [the challenge flow section](#integrate-sdk-for-authenticator-challenge).
+
+The names of the authenticators available for enrollment or challenge can be found in the `AuthenticationResponse` object's `authenticators` collection. Display a list of authenticators to select the Google Authenticator for enrollment.
+
+> **Note:** The `isChallengeFlow` session variable is set to `false` if the user needs to enroll Google Authenticator first, and `true` if they have already done so.
+
+```java
+case AWAITING_AUTHENTICATOR_ENROLLMENT_SELECTION:
+
+    // There are multiple potential factors, for this example we are going to
+    // assume Google Authenticator (OTP) is used
+    Authenticator authenticator = authenticationResponse.getAuthenticators().stream()
+            .filter(auth -> auth.getType().equals("otp"))
+            .findFirst() // find the first "otp" factor or throw an exception
+            .orElseThrow(() -> {
+                String availableFactors = authenticationResponse.getAuthenticators().stream()
+                        .map(auth -> auth.getLabel() + " - " + auth.getType())
+                        .collect(Collectors.joining(", "));
+                return new RuntimeException("OTP factor not found, existing options: " + availableFactors);
+            });
+
+    // Tell Okta which factor was selected, and go back into the state machine with the new response
```

Comment:
The two clauses say much the same thing. How about:
// Update the response with the selected authenticator and send the result back to the server.